nginx/nginx.go

```diff
@@ -448,6 +438,42 @@ func createServerEntries(update controller.IngressUpdate) []*server {
 	return serverEntries
 }
 
+type ingressKey struct {
+	Host, Path string
+}
+
+func uniqueIngressEntries(ingressEntries []controller.IngressEntry) []controller.IngressEntry {
+	uniqueIngress := make(map[ingressKey]controller.IngressEntry)
+	for _, ingressEntry := range ingressEntries {
+		key := ingressKey{ingressEntry.Host, ingressEntry.Path}
+		existingIngressEntry, exists := uniqueIngress[key]
+		if !exists {
+			uniqueIngress[key] = ingressEntry
+			continue
+		}
+		mostRecentIngress, olderIngress := orderByCreation(ingressEntry, existingIngressEntry)
+		log.Infof("Ignoring '%s' because it duplicates the host/path of a more recent entry '%s'", olderIngress.NamespaceName(), mostRecentIngress.NamespaceName())
+		uniqueIngress[key] = mostRecentIngress
+	}
+
+	uniqueIngressEntries := make([]controller.IngressEntry, 0, len(uniqueIngress))
+	for _, value := range uniqueIngress {
+		uniqueIngressEntries = append(uniqueIngressEntries, value)
+	}
+	return uniqueIngressEntries
+}
+
+func orderByCreation(entry1, entry2 controller.IngressEntry) (mostRecentIngress, olderIngress controller.IngressEntry) {
```

Comment:
For future reference don't use named return parameters, it's considered bad practice in golang.